types/aqb/index.d.ts

```diff
@@ -0,0 +1,1773 @@
+     *
+     * Examples
+     *
+     * qb.obj({'some.name': 'value'}) :  {"some.name": value}
+     *
+     * qb.obj({hello: world}) :  {hello: world}
+     *
+     * qb.obj({'"hello"': world}) :  {"hello": world}
+     *
+     * qb.obj({':dynamic': 'props'}) :  {[dynamic]: props}
+     *
+     * qb.obj({': invalid': 'key'}) :  throws an error (invalid is not a well-formed reference)
+     *
+     * @param obj
+     */
+    obj (obj: {}) :  ObjectLiteral;
+    /**
+     * Simple Reference
+     *
+     * Wraps a given value in an  Simple Reference.
+     *
+     * qb.ref(value)
+     *
+     * If the value is not a JavaScript string or not a well-formed simple reference, an Error will be thrown.
+     * If the value is an ArangoCollection, its name property will be used instead.
+     * If the value is already an  Simple Reference, its value is wrapped instead.
+     *
+     * Examples
+     *
+     * Valid values:
+     *
+     * foo
+     *
+     * foo.bar
+     *
+     * foo[*].bar
+     *
+     * foo.bar.QUX
+     *
+     * _foo._bar._qux
+     *
+     * foo1.bar2
+     *
+     * `foo`.bar
+     *
+     * foo.`bar`
+     *
+     * Invalid values:
+     *
+     * 1foo
+     *
+     * föö
+     *
+     * foo bar
+     *
+     * foo[bar]
+     *
+     * @param value
+     * @return
+     */
+    ref (value: string) :  SimpleReference;
+    /**
+     * Expressions
+     *
+     * @param value
+     */
+    expr (value: any) :  RawExpression;
+    /**
+     * Ternary (if / else)
+     *
+     * Creates a ternary expression from the given values.
+     *
+     * qb.if(condition, thenDo, elseDo) :  (condition ? thenDo : elseDo)
+     * OR:
+     * qbValue.then(thenDo).else(elseDo) :  (condition ? thenDo : elseDo)
+     *
+     * If the values are not already  values, they will be converted automatically.
+     *
+     * Alias: qbValue.then(thenDo).otherwise(elseDo)
+     *
+     * Examples
+     * qb.ref('x').then('y').else('z') :  (x ? y : z)
+     *
+     * @param cond
+     * @param then
+     * @param otherwise
+     */
+    if (cond: any, then: any, otherwise: any) :  Expression | number;
+    /**
+     * Function Call
+     *
+     * Creates a functon call for the given name and arguments.
+     *
+     * qb.fn(name)(...args)
+     *
+     * If the values are not already  values, they will be converted automatically.
+     * For built-in functions, methods with the relevant function name are already provided by the query builder.
+     *
+     * Examples
+     *
+     * qb.fn('MY::USER::FUNC')(1, 2, 3) :  MY::USER::FUNC(1, 2, 3)
+     *
+     * qb.fn('hello')() :  hello()
+     *
+     * qb.RANDOM() :  RANDOM()
+     *
+     * qb.FLOOR(qb.div(5, 2)) :  FLOOR((5 / 2))
+     *
+     * @param functionName
+     * @param arity
+     * @return
+     */
+    fn (functionName: string) :  (...arity: any[]) =>  FunctionCall;
+}
+
+export declare type QBfunc = (obj : any) => functions
+export declare const QB : functions & QBfunc
+
+export default QB
```

Comment:
This should be `export = QB` and not `export default QB`, to [align with the shape the library actually exports](https://github.com/arangodb/aqbjs/blob/master/index.js#L124). If at compile time you expect your module loader will make a commonjs module available on the default member of the import, you should run TS with the flag `--allowSyntheticDefaultImports`, or (with the upcoming TS 2.7) `--esModuleInterop`. If not, it should _probably_ be imported with the `import QB = require("aqb")` syntax, since the shape can't be callable or construct-able as an es6 namespace import.

Reply:
I have only been doing typescript for a few months. This proved to be very tricky because of the manner in which the library is written. Will implement your suggestion. However I personally prefer developers to access all the interfaces of a library.